### PR TITLE
Feature/pn 1429 notification cost

### PR DESCRIPTION
**Pull request by masciale89**

### CODE REVIEW

File: scripts/aws/cfn/microservice.yml [hunk 1]
```diff
@@ -58,6 +58,14 @@ Parameters:
     Type: String
     Description: 'ARN of dynamodb table containing notifications'
 
+  NotificationsCostDynamoTableName:
+    Type: String
+    Description: 'Notifications Cost Dynamo table name'
+
+    NotificationsCostDynamoTableArn:
```
Comment (marco-vit-pagopa) [inline scripts/aws/cfn/microservice.yml:65]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Parameter NotificationsCostDynamoTableName has invalid property NotificationsCostDynamoTableArn

**Review by MTurraPpa — APPROVED** [review]

File: src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java [hunk 2]
```diff
@@ -24,7 +27,20 @@ public PnNotificationPriceController(NotificationPriceService service) {
...
+                .mdcEntry("paTaxId",paTaxId)
+                .mdcEntry("noticeNumber", noticeNumber)
+                .build();
+        NotificationPriceResponse response = new NotificationPriceResponse();
+        try {
+            response = service.getNotificationPrice( paTaxId, noticeNumber );
+            logEvent.generateSuccess().log();
+        } catch (PnNotFoundException exc) {
+            logEvent.generateFailure("Exception on get notification price: " + exc.getMessage()).log();
+            throw new RuntimeException("Exception on get notification price: " + exc.getMessage(), exc);
```
Comment (MTurraPpa) [inline src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java:42]: Perché trasformare una PnNotFoundException che dovrebbe essere una exception gestita in una generica RuntimeException?

**Review by MTurraPpa — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public ResponseEntity<NotificationPriceResponse> getNotificationPrice(String paT
             logEvent.generateSuccess().log();
         } catch (PnNotFoundException exc) {
             logEvent.generateFailure("Exception on get notification price: " + exc.getMessage()).log();
-            throw new RuntimeException("Exception on get notification price: " + exc.getMessage(), exc);
+            throw new PnNotFoundException("Exception on get notification price: " + exc.getMessage(), exc);
         }
```
Comment (MTurraPpa) [inline src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java:43]: Qui fa la catch di PnNotFoundException e va rilanciata la stessa, no importa crearne una nuova. Ma ci sono anche le altre Exception che devono comunque finire nell'audit log.

**Review by MTurraPpa — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java (hunk 3 above)
Comment (cFiorilloMediatica) [reply src/main/java/it/pagopa/pn/delivery/rest/PnNotificationPriceController.java:43]: Qui mi sono ispirato alla documentazione IBM https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions sulle best-practice, in particolare il punto 4.

> The correct way to catch and re-throw an exception is to pass the caught exception object as the "rootCause" or inner exception parameter to the constructor of the new exception (note that not all exception constructors support inner exceptions, in which case a different wrapper exception should be used). When the exception is later caught and printed to SystemOut.log, the inner exception message and stack will be included:

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -7,6 +7,11 @@ pn.delivery.download-url-duration=5m
 pn.delivery.download-with-presigned-url=true
 pn.delivery.number-of-presigned-request=15
 pn.delivery.max-page-size=4
+pn.delivery.costs.notification=200
+pn.delivery.costs.raccomandata-ita=540
+pn.delivery.costs.raccomandata-est-zona1=710
+pn.delivery.costs.raccomandata-est-zona2=840
+pn.delivery.costs.raccomandata-est-zona3=905
```
Comment (MTurraPpa) [inline src/main/resources/application.properties:15]: Queste sono replicate in config, non bastano qua?

**Review by MTurraPpa — APPROVED** [review]

**Review by MTurraPpa — APPROVED** [review]

**Pull request merged**